showcase-v1alpha2/src/v1alpha2/echo_client.ts

```diff
@@ -0,0 +1,688 @@
+   *   .then(callback)
+   *   .catch(err => {
+   *     console.error(err);
+   *   });
+   */
+  pagination(
+      request: PaginationRequest, options?: gax.CallOptions,
+      callback?: GapicPaginationCallback<
+          PaginationRequest, PaginationResponse, number>):
+      Promise<GapicPaginationResponse<
+          PaginationRequest, PaginationResponse, number>> {
+    options = options || {};
+    // Here we receive an untyped array from gax and convert it to typed object.
+    return new Promise((resolve, reject) => {
+      this._innerApiCalls.pagination(request, options, callback)
+          .then((resultArray: Array<{}>) => {
+            const result: GapicPaginationResponse<
+                PaginationRequest, PaginationResponse, number> = {
+              values: resultArray[0] as number[],
+              nextPageRequest: resultArray[1] as PaginationRequest,
+              rawResponse: resultArray[2] as PaginationResponse
+            };
+            resolve(result);
+          })
+          .catch(reject);
+    });
+  }
+
+  /**
+   * Equivalent to {@link pagination}, but returns a NodeJS Stream object.
+   *
+   * This fetches the paged responses for {@link pagination} continuously
+   * and invokes the callback registered for 'data' event for each element in
+   * the responses.
+   *
+   * The returned object has 'end' method when no more elements are required.
+   *
+   * autoPaginate option will be ignored.
+   *
+   * @see {@link https://nodejs.org/api/stream.html}
+   *
+   * @param {Object} request
+   *   The request object that will be sent.
+   * @param {number} request.maxResponse
+   *   The maximum number that will be returned in the response.
+   * @param {number} [request.pageSize]
+   *   The maximum number of resources contained in the underlying API
+   *   response. If page streaming is performed per-resource, this
+   *   parameter does not affect the return value. If page streaming is
+   *   performed per-page, this determines the maximum number of
+   *   resources in a page.
+   * @param {Object} [options]
+   *   Optional parameters. You can override the default settings for this call,
+   * e.g, timeout, retries, paginations, etc. See [gax.CallOptions]{@link
+   * https://googleapis.github.io/gax-nodejs/global.html#CallOptions} for the
+   * details.
+   * @returns {Stream}
+   *   An object stream which emits a number on 'data' event.
+   *
+   * @example
+   *
+   * const showcase = require('showcase.v1alpha2');
+   *
+   * const client = new showcase.v1alpha2.EchoClient({
+   *   // optional auth parameters.
+   * });
+   *
+   * const maxResponse = 0;
+   * client.paginationStream({maxResponse: maxResponse})
+   *   .on('data', element => {
+   *     // doThingsWith(element)
+   *   }).on('error', err => {
+   *     console.log(err);
+   *   });
+   */
+  paginationStream(request: PaginationRequest, options?: gax.CallOptions):
+      NodeJS.ReadableStream {
```

Comment:
This separate method is included only if the generator thinks pagination is possible? And the `pagination()` method is unaltered?

Reply:
Yes, this is how it works with gapic-generator.

https://github.com/googleapis/gapic-generator/blob/ded2082dee6669b6a79ea72e0ecc33fafecd42e4/src/main/resources/com/google/api/codegen/nodejs/main.snip#L477-L508